src/modules/class/application/class.service.ts

```diff
@@ -0,0 +1,43 @@
+
+import { Injectable, Inject } from "@nestjs/common";
+import { IClassRepository, CLASS_REPOSITORY_TOKEN } from "../domain/class-repository.interface";
+import { CreateClassDTO } from "./create-class.dto";
+import { Class } from "../domain/class.entity";
+
+@Injectable()
+export class ClassService {
+    constructor(
+        @Inject(CLASS_REPOSITORY_TOKEN)
+        private readonly classRepository: IClassRepository,
+    ) {}
+
+    async createClass(createClassDto: CreateClassDTO): Promise<Class> {
+        const classEntity = new Class({
+            name: createClassDto.name,
+            activityId: createClassDto.activityId,
+            levelId: createClassDto.levelId,
+            state: createClassDto.state,
+            studentsIds: createClassDto.studentsIds || [],
+            teacherIds: createClassDto.teacherIds || [],
+        });
+
+        return await this.classRepository.create(classEntity);
+    }
+
+    async findById(id: number): Promise<Class | null> {
+        return await this.classRepository.findById(id);
+    }
```

Comment:
```suggestion
    async findById(id: number): Promise<Class> {
        var _class = await this.classRepository.findById(id);
        if (!_class) {
        throw new NotFoundException(`Class with ID ${id} not found`);
      }
    }
```
É interessante que as exceções sejam lançadas na camada aplicação, não no repository ou na controller.